src/main/java/hudson/plugins/xvnc/Xvnc.java

```diff
@@ -139,20 +145,56 @@ private void doSetUp(Context context, Run<?,?> build, FilePath workspace, Node n
         }
 
         context.env("DISPLAY", ":" + displayNumber);
-        context.setDisposer(new DisposerImpl(displayNumber, xauthorityEnv, vncserverCommand, takeScreenshot, xauthority.getRemote()));
+        context.setDisposer(new DisposerImpl(displayNumber, xauthorityEnv, vncserverCommand, takeScreenshot, xauthority != null ? xauthority.getRemote() : null));
     }
-    
+
+    /**
+     * Attempts to find a suitable place for the Xauthority file where the path to it doesn't contain any spaces.
+     * The order of tries is the workspace, the slave's fs root and last the system temp dir.
+     * If the system temp dir also contains a space a warning will be printed to the log but the temp dir path will
+     * be created and returned anyways.
+     * @param workspace the build's workspace.
+     * @param logger the build's log to print the warning to.
+     * @return the path on the slave to the created temp file.
+     *
+     * @throws IOException if so
+     * @throws InterruptedException if so
+     */
+    private FilePath createXauthorityFile(FilePath workspace, final PrintStream logger) throws IOException, InterruptedException {
+        if (workspace.getRemote().indexOf(' ') < 0) {
+            //If the workspace doesn't have any spaces it is probably safe
+            return workspace.createTempFile(".Xauthority-", "");
+        } else {
+            //Try the fs root
+            Computer computer = workspace.toComputer();
+            Node node = computer.getNode();
+            FilePath rootPath = node.getRootPath();
+            if (rootPath.getRemote().indexOf(' ') < 0) {
+                return rootPath.createTempFile(".Xauthority-", "");
+            } else {
+                FilePath file = workspace.createTextTempFile(".Xauthority-", "", "", false);
```

Comment:
this is insecure - needs to use java7 to create this securely.  Most likely ok as a slave is generally not a multi user machine - but probably worth a follow up //TODO